### PR TITLE
Adding Exception for models to raise if initialization fails

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
Currently, initialization routines have no way of telling the user that they failed beyond logger messages. This PR adds a new InitializationError that all initialization routines should raise if there is a failure during initiation (e.g. the final solve is not feasible).

## Changes proposed in this PR:
- Adds new InitializaitonError
- Implements code to raise InitializationErrors in most core property packages.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/util/exceptions.py [hunk 1]
```diff
@@ -66,3 +66,13 @@ class PropertyPackageError(AttributeError):
     Needs to inherit from AttributeError for Pyomo interactions.
     """
     pass  # Bread stuck
+
+
+class InitializationError(ValueError):
```
Comment (jsiirola) [inline idaes/core/util/exceptions.py:71]: Is ValueError the correct base class?

File: idaes/generic_models/properties/tests/test_iapws95_with_harness.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def configure(self):
         self.param_args = {"phase_presentation": iapws95.PhaseType.MIX}
         self.prop_args = {}
         self.has_density_terms = True
+        # Helmholt package initialization has not solver calls, so can't fail
```
Comment (jsiirola) [inline idaes/generic_models/properties/tests/test_iapws95_with_harness.py:30]: Typos: :Helmholt" -> "Helmholtz", "not" -> "no"

(here and 3 copy/paste locations below)

File: idaes/generic_models/properties/tests/test_harness.py [hunk 3]
```diff
@@ -51,11 +52,14 @@ def configure_class(self, m):
 
         self.has_density_terms = True
 
+        self.skip_initialization_exception = False
```
Comment (jsiirola) [inline idaes/generic_models/properties/tests/test_harness.py:55]: Would it be more clear if this were `skip_initialization_raises_exception_test`?

File: idaes/generic_models/properties/tests/test_swco2_with_harness.py [hunk 4]
```diff
@@ -27,6 +27,8 @@ def configure(self):
         self.param_args = {"phase_presentation": swco2.PhaseType.MIX}
         self.prop_args = {}
         self.has_density_terms = True
+        # Helmholt package initialization has not solver calls, so can't fail
```
Comment (jsiirola) [inline idaes/generic_models/properties/tests/test_swco2_with_harness.py:30]: Typos: :Helmholt" -> "Helmholtz", "not" -> "no"

(here and 3 copy/paste locations below)

File: idaes/core/util/exceptions.py [hunk 5]
```diff
@@ -66,3 +66,13 @@ class PropertyPackageError(AttributeError):
...
+    """
+    IDAES exception to be used when initialization routines fail. All
+    initialization routines should raise this Exception if the final step
+    fails ot converge, and can raise this exception earlier if the routine
```
Comment (blnicho) [inline idaes/core/util/exceptions.py:75]: Typo: to

File: idaes/generic_models/properties/core/generic/utility.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ class StateIndex(Enum):
     apparent = 2
 
 
-class GenericPropertyPackageError(PropertyPackageError):
+class GenericPropertyPackageError(PropertyPackageError, IdaesError):
```
Comment (jsiirola) [inline idaes/generic_models/properties/core/generic/utility.py:35]: You don't need to list `IdaesError` here, as it is already in the class hierarchy through `PorpertyPackageError`

**Review by jsiirola — COMMENTED** [review]
One nit, then I think this is ready.

File: idaes/core/util/exceptions.py [hunk 7]
```diff
@@ -18,38 +18,47 @@
 __author__ = "Andrew Lee"
 
 
-class BalanceTypeNotSupportedError(NotImplementedError):
+class IdaesError(Exception):
     """
-    IDAES exception to be used when a control volumedoes not support a given
+    General exception for all IDAES errors. Intended to be used for multiple
+    inheritance in derived expcetions to allow catching of all IDAES-related
```
Comment (blnicho) [inline idaes/core/util/exceptions.py:24]: Typo: exceptions

**Review by jsiirola — APPROVED** [review]

**Review by blnicho — APPROVED** [review]

**Pull request merged**